ci/rust-version.sh

```diff
@@ -46,6 +46,7 @@ export rust_nightly_docker_image=solanalabs/rust-nightly:"$nightly_version"
     if ! cargo +"$toolchain" -V > /dev/null; then
       echo "$0: Missing toolchain? Installing...: $toolchain" >&2
       rustup install "$toolchain"
+      rustup +"$toolchain" component add clippy rustfmt
       cargo +"$toolchain" -V
     fi
```

Comment:
actually, I don't like this whole `if` block to begin with.... I've bitten by this numerous times. i think altering some system state outside `git rev-parse --show-toplevel` is too obstructive/unsuitable. (i.e. `rustup {install/component add}` without manual intervention).

That aside, i think these list of component could get stale easily.. so, I'd like this to be sourced into here to be shared with `Dockerfile`s. good-old plain text file with few lines should suffice.

@yihau what do you think? iirc, you have some envisioned idea for our ci rust docker images?

Reply:
yeah. add this one to our Dockerfile or Ansible file (will open the repo soon!) will be better.
btw, my idea is like to wrap everything into custom images which will be tagged by branch. the goal is to make everyone can test them easily. I put this thing after I refactoring pipeline.

Reply:
Your call - I'm going to close it.
But just to point out, I do not think that
```
rustup +"$toolchain" component add clippy rustfmt
```
will get stale any time soon.
Rust was packaging these tools with these names for at least 6 years now.
And it is unlikely they are going to change.

I do agree that if you can manage your system state as code - that is ideal.
But it requires corresponding tools.
IIUC, Nix is the best in class.
Bazel and Buck being the close second.
Dockerfiles have issues with reproducibility.

At the same time, rust toolchains and component have very stable API.
Toolchains are versioned, and component names are very unlikely to ever change.